From 283b10b15db65993d842d5716284b4cb52187498 Mon Sep 17 00:00:00 2001
From: Dave Allured <dave.allured@noaa.gov>
Date: Thu, 10 Dec 2020 19:38:40 -0700
Subject: [PATCH 1/3] NclApi.c: Change 3 global vars to extern.

Fix for duplicate symbol errors with gcc10.
Base definition for these 3 globals are in NclGlobalVars.h.
This is better than -fcommon.
---
 ni/src/ncl/NclApi.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/ni/src/ncl/NclApi.c b/ni/src/ncl/NclApi.c
index c5b1f926d..5efe140c3 100644
--- a/ni/src/ncl/NclApi.c
+++ b/ni/src/ncl/NclApi.c
@@ -140,9 +140,10 @@ extern char *the_input_buffer;
 extern char *the_input_buffer_ptr;
 extern int the_input_buffer_size;
 
-FILE *thefptr;
-FILE *theoptr;
-int cmd_line;
+extern FILE *thefptr;
+extern FILE *theoptr;
+extern int cmd_line;
+
 extern int cur_line_number;
 extern char *cur_line_text;
 extern int cur_line_maxsize;

From ac228898aa25273e25c84083b12cb46e16b54f5c Mon Sep 17 00:00:00 2001
From: Dave Allured <dave.allured@noaa.gov>
Date: Thu, 10 Dec 2020 19:47:23 -0700
Subject: [PATCH 2/3] NclHdf5.c: Change 1 global var to static

Fix for duplicate symbol error with gcc10.
This one is local use only, and should never have been global.
---
 ni/src/ncl/NclHDF5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ni/src/ncl/NclHDF5.c b/ni/src/ncl/NclHDF5.c
index ddaf791b9..718890e3a 100644
--- a/ni/src/ncl/NclHDF5.c
+++ b/ni/src/ncl/NclHDF5.c
@@ -240,7 +240,7 @@ struct _HDF5FileRecord
 
 #define NUMPOSDIMNAMES	6
 
-NclQuark possibleDimNames[NUMPOSDIMNAMES];
+static NclQuark possibleDimNames[NUMPOSDIMNAMES];
 
 static int _H5_initializeOptions 
 #if    NhlNeedProto

From 089e1fec80bd40424b2782f5cfb3102215d2a683 Mon Sep 17 00:00:00 2001
From: Dave Allured <dave.allured@noaa.gov>
Date: Thu, 10 Dec 2020 19:50:23 -0700
Subject: [PATCH 3/3] NclNewHdf5.c: Change 1 global var to static

Fix for duplicate symbol error with gcc10.
This one is local use only, and should never have been global.
---
 ni/src/ncl/NclNewHDF5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ni/src/ncl/NclNewHDF5.c b/ni/src/ncl/NclNewHDF5.c
index 9cd5eda48..f246324eb 100644
--- a/ni/src/ncl/NclNewHDF5.c
+++ b/ni/src/ncl/NclNewHDF5.c
@@ -82,7 +82,7 @@ static NrmQuark Qfill_val;
 
 #define NUMPOSDIMNAMES	6
 
-NclQuark possibleDimNames[NUMPOSDIMNAMES];
+static NclQuark possibleDimNames[NUMPOSDIMNAMES];
 
 #ifndef FALSE
 #define FALSE           0
